kratos/input_output/vtk_output.cpp

```diff
@@ -431,21 +449,30 @@ void VtkOutput::WriteConditionResultsToFile(const ModelPart& rModelPart, std::of
 {
     const auto& r_local_mesh = rModelPart.GetCommunicator().LocalMesh();
     Parameters condition_results = mOutputSettings["condition_data_value_variables"];
+    Parameters condition_flags = mOutputSettings["condition_flags"];
 
     int num_elements = static_cast<int>(r_local_mesh.NumberOfElements());
     rModelPart.GetCommunicator().SumAll(num_elements);
 
     int num_conditions = static_cast<int>(r_local_mesh.NumberOfConditions());
     rModelPart.GetCommunicator().SumAll(num_conditions);
 
-    if (num_elements == 0 && num_conditions > 0) {
-        // write cells header
+    if (num_elements == 0 && num_conditions > 0) { // TODO: Can we have conditions and elements at the same time?
+        // Write cells header
         rFileStream << "CELL_DATA " << r_local_mesh.NumberOfConditions() << "\n";
-        rFileStream << "FIELD FieldData " << condition_results.size() << "\n";
+        rFileStream << "FIELD FieldData " << condition_results.size() + condition_flags.size() << "\n";
         for (IndexType entry = 0; entry < condition_results.size(); ++entry) {
             const std::string& r_condition_result_name = condition_results[entry].GetString();
             WriteGeometricalContainerResults(r_condition_result_name,r_local_mesh.Conditions(),rFileStream);
         }
+
+        // Writing condition_flags
+        for (IndexType entry = 0; entry < condition_flags.size(); ++entry) {
+            // write nodal results variable header
+            const std::string& r_condition_result_name = condition_flags[entry].GetString();
+            const Flags flag = KratosComponents<Flags>::Get(r_condition_result_name);
```

Comment:
just to say that we will only be able to write Flags that are defined in the Core
App-Flags will not be able to be written since they are not registered.

@jcotela am I wrong here?

Reply:
If you are loading the applications other flags should appear

Reply:
hm I thought the same but I tried this in the IGAApp and there we could not retrieve the local iga flags from the Globals 
I can try again though in the scope of this PR

I mean, it does not affect this PR, there is not much we can do if it is not working

Reply:
I understand that application flags should be available as @loumalouomega says (but I never tested it). What we won't have is class-specific flags (such as those defined by the IO class or the TetrahedralMeshOrientationCheck, for example), which are not registered.